### PR TITLE
docs: section about dropped openapi option

### PR DESCRIPTION
**Pull request by georgettica**
Because `operator-sdk generate openapi` has been dropped, it is good to add here a section on how to work with openapi from now on

**Description of the change:**

**Motivation for the change:**

### CODE REVIEW

**Comment by georgettica** [comment]
@camilamacedo86  @joelanford I created this PR

File: website/content/en/docs/faq.md [hunk 1]
```diff
@@ -142,6 +142,10 @@ $ export GOROOT=$(go env GOROOT)
 
 This will work for the current environment. To persist this fix, add the above line to your environment's config file, ex. `bashrc` file.
 
+# I cannot use the command `operator-sdk generate openapi`
```
Comment (camilamacedo86) [inline website/content/en/docs/faq.md:145]:
```suggestion
## I cannot use the command `operator-sdk generate openapi`
```

File: website/content/en/docs/faq.md [hunk 2]
```diff
@@ -142,6 +142,10 @@ $ export GOROOT=$(go env GOROOT)
...
+A decision was made to extract this tool outside of operator-sdk in version v0.17, see migration guide on for that version [here][version_17x] and you should know that file file `./hack/boilerplate.go.txt` is for adding a LICENCE file to the top of documents, and can be removed
```
Comment (camilamacedo86) [inline website/content/en/docs/faq.md:146]:
```suggestion
The command `operator-sdk generate openapi` was removed from the SDK tool in the version `v0.17.0`. It is now recommended to use [openapi-gen][openapi-gen] directly for OpenAPI code generation. 

Note that the file `./hack/boilerplate.go.txt` is used to allow add the LICENCE comment on the top of the documents and then, it can be removed.
```

File: website/content/en/docs/faq.md [hunk 3]
```diff
@@ -150,3 +154,5 @@ This will work for the current environment. To persist this fix, add the above l
 [client.Reader]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#Reader
 [rbac]:https://kubernetes.io/docs/reference/access-authn-authz/rbac/
 [goroot-github-issue]:https://github.com/operator-framework/operator-sdk/issues/1854#issuecomment-525132306
+[version_17x]: /docs/migration/version-upgrade-guide/#v017x
```
Comment (camilamacedo86) [inline website/content/en/docs/faq.md:157]:
```suggestion
[openapi-gen]: https://github.com/kubernetes/kube-openapi/tree/master/cmd/openapi-gen
```

File: website/content/en/docs/faq.md (hunk 1 above)
Comment (camilamacedo86) [reply website/content/en/docs/faq.md:145]: To fix the layout.

File: website/content/en/docs/faq.md (hunk 2 above)
Comment (camilamacedo86) [reply website/content/en/docs/faq.md:146]: We can point out the users to the right place instead of using the migration guide.

**Comment by georgettica** [comment]
did all the changes you requested @camilamacedo86, can you take a look?

File: website/content/en/docs/faq.md [hunk 4]
```diff
@@ -142,6 +142,12 @@ $ export GOROOT=$(go env GOROOT)
 
 This will work for the current environment. To persist this fix, add the above line to your environment's config file, ex. `bashrc` file.
 
+## I cannot use the command `operator-sdk generate openapi`
+The command `operator-sdk generate openapi` was removed from the SDK tool in the version `v0.17.0`. It is now recommended to use [openapi-gen][openapi-gen] directly for OpenAPI code generation.
```
Comment (joelanford) [inline website/content/en/docs/faq.md:146]: Can we reword this to something like:

```suggestion
## How do I generate OpenAPI Go source files for my APIs?
It is recommended to use [openapi-gen][openapi-gen] for OpenAPI code generation.
```

File: website/content/en/docs/faq.md [hunk 5]
```diff
@@ -142,6 +142,12 @@ $ export GOROOT=$(go env GOROOT)
 
 This will work for the current environment. To persist this fix, add the above line to your environment's config file, ex. `bashrc` file.
 
+## I cannot use the command `operator-sdk generate openapi`
+The command `operator-sdk generate openapi` was removed from the SDK tool in the version `v0.17.0`. It is now recommended to use [openapi-gen][openapi-gen] directly for OpenAPI code generation. 
+
+Note that in the example on the docs, the flag `-h ./hack/boilerplate.go.txt` is used to allow add the LICENCE comment on the top of the documents and then, it can be removed.
```
Comment (joelanford) [inline website/content/en/docs/faq.md:148]: Where is the example? I think we should embed that directly in the FAQ.

Note that we're switching to the kubebuilder-based project layout in #3190, so we should make the example work for that project structure (not our existing project structure).

@hasbro17 did you run across an examples in our docs for running openapi-gen that had to change due to the new layout?

Comment (hasbro17) [reply website/content/en/docs/faq.md:148]: @joelanford Hmm no. The only place where we mention how to use openapi-gen is in the version upgrade guide right around the time we dropped support for `operator-sdk generate openapi`.
https://sdk.operatorframework.io/docs/migration/version-upgrade-guide/#v017x

But I guess for the new project layout you'd just need to update the input/output package paths to your API Go types and you can use that:
```console
# Install openapi-gen binary. Although not this way since this pollutes your SDK's go.mod file
$ go get k8s.io/kube-openapi/cmd/openapi-gen

$ tree apis/
apis/
└── cache
    └── v1alpha1
        ├── groupversion_info.go
        ├── memcached_types.go
        └── zz_generated.deepcopy.go

# openapi-gen always needs a go header file so just use an empty one if not needed
$ touch header.txt

$ openapi-gen  --output-base="" --input-dirs=./apis/cache/v1alpha1 --output-package=./apis/cache/v1alpha1 --output-file-base=zz_generated.openapi -h=./header.txt

$ tree apis/
apis/
└── cache
    └── v1alpha1
        ├── groupversion_info.go
        ├── memcached_types.go
        ├── zz_generated.deepcopy.go
        └── zz_generated.openapi.go
```

Oh also, with Kubebuilder we don't have the `// +k8s:openapi-gen=true` tags in our `<kind>_type.go` scaffolds anymore. So you'd probably need to add those to the Go types yourself.

Comment (camilamacedo86) [reply website/content/en/docs/faq.md:148]: Hi @georgettica could you please check it with the new layout to provide the example? To do that you need use the master branch and run `mkdir <projectname>`, `cd <projectname>` and then, `operator-sdk init`

Comment (georgettica) [reply website/content/en/docs/faq.md:148]: will it skaffold a current project with the new information? for I got a confusing result.
I ran the command on my current operator and recieved an error but files and folders were created.

When I ran the init command a `hack/boilerplate.go.txt` was created so if I were using the command in https://sdk.operatorframework.io/docs/migration/version-upgrade-guide/#v017x

and I guess the tags`// +k8s:openapi-gen=true` were removed for extracting the openapi-gen from the codebase, so a not should refer to that

Comment (camilamacedo86) [reply website/content/en/docs/faq.md:148]: Hi @georgettica,

SDK is in a process to be integrated with KB which means that its project layouts will be aligned. More info : [Integrating Kubebuilder and Operator SDK](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/integrating-kubebuilder-and-osdk.md).  If you run `operator-sdk init` you will scaffold a project with the new layout. 

What is missing here is we provide the example and the info let the users know how to do with the Legacy layout and the new one. Is it make sense?

**Comment by camilamacedo86** [comment]
Hi @georgettica, 

I am closing this one since it is open for a while without address the suggestion.
However, feel free to re-open this one with you wish and if you be able to address the requests changes to add info requested. 

Really thank you for your contribution. 
And please you are more than welcome to push PR's and contribute to this project.
